DTML.EduBot/Common/TraceSourceExceptionLogger.cs

```diff
@@ -0,0 +1,31 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using System.Web;
+using System.Web.Http.ExceptionHandling;
+
+namespace DTML.EduBot.Common
+{
+    public class TraceSourceExceptionLogger : ExceptionLogger
+    {
+        private readonly TraceSource _traceSource;
+
+        public TraceSourceExceptionLogger(TraceSource traceSource)
+        {
+            _traceSource = traceSource;
+        }
+
+        public override void Log(ExceptionLoggerContext context)
+        {
+            _traceSource.TraceEvent(TraceEventType.Error, 1,
+                "Unhandled exception processing {0} for {1}: {2}, w/ user ID: {3}",
+                context.Request.Method,
+                context.Request.RequestUri,
+                context.Exception,
+                context.Exception.Data["id"]);
+
+            Console.WriteLine(context.Exception.Data["id"]);
```

Comment:
Oops, forgot to take out that line :P Otherwise, this was meant to just check in a simple skeleton for global error handling